op-batcher/txmgr.go

```diff
@@ -2,13 +2,56 @@ package op_batcher
 
 import (
 	"context"
+	"time"
 
 	"github.com/ethereum-optimism/optimism/op-proposer/txmgr"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
+func (l *BatchSubmitter) submitTransaction(data []byte) error {
+	// Query for the submitter's current nonce.
+	ctx, cancel := context.WithTimeout(l.ctx, time.Second*10)
+	nonce, err := l.cfg.L1Client.NonceAt(ctx, l.addr, nil)
+	cancel()
+	if err != nil {
+		l.log.Error("unable to get current nonce", "err", err)
+		return err
+	}
+
+	// Create the transaction
+	ctx, cancel = context.WithTimeout(l.ctx, time.Second*10)
+	tx, err := l.CraftTx(ctx, data, nonce)
+	cancel()
+	if err != nil {
+		l.log.Error("unable to craft tx", "err", err)
+		return err
+	}
+
+	// Construct the a closure that will update the txn with the current gas prices.
+	updateGasPrice := func(ctx context.Context) (*types.Transaction, error) {
+		l.log.Debug("updating batch tx gas price")
+		return l.UpdateGasPrice(ctx, tx)
+	}
+
+	// Wait until one of our submitted transactions confirms. If no
+	// receipt is received it's likely our gas price was too low.
+	// TODO: does the tx manager nicely replace the tx?
```

Comment:
The tx manager will replace the transaction until the resubmission timeout is reached. The resubmission timeout is separate from the channel timeout, though, so we'll want to make sure those are in sync.